### PR TITLE
fix(Checkbox): Update checkbox styles

### PR DESCRIPTION
**Pull request by kosanna**
- Add activeRoot demo
- Make active and focus states be the same on checkbox
- Organise active/focused state reset to a mixin
- Make checkbox background colour always white
- Make tick be the same colour as focus/active outline with potential opt-out from them being the same

### CODE REVIEW

File: react/Checkbox/Checkbox.less [hunk 1]
```diff
@@ -40,16 +40,16 @@
...
+    background-color: @sk-white;
   }
 }
 
 .checkMark {
   display: block;
   width: 100%;
   height: 100%;
-  fill: @sk-black;
+  fill: @sk-focus;
```
Comment (markdalgleish) [inline react/Checkbox/Checkbox.less]: This isn't really semantically correct, because if we tweak our focus styles, it would be unexpected to *also* be changing the colour of a tick. We might need to introduce another variable here, like `@sk-form-accent`?

Comment (kosanna) [inline react/Checkbox/Checkbox.less]: The point of this change was to make them aligned. So it should be sk-focus in case we do change focus color they are still aligned.

Comment (markdalgleish) [inline react/Checkbox/Checkbox.less]: Can probably leave this as-is for now, but this will probably need to trigger a design discussion about whether our focus colours should be coupled to the form element design in this way, cc @stephaniemoss

Comment (michaeltaranto) [inline react/Checkbox/Checkbox.less]: That's fine if they are deliberately in sync at the moment. But going with `@sk-form-accent` future proofs the api. You can set `@sk-form-accent: @sk-focus` to allow the flexibility to change this independently later.

Comment (kosanna) [inline react/Checkbox/Checkbox.less]: Good point @michaeltaranto thought about that. We already had the design discussion re that and it was how I got it, so they always should be in sync...

**Comment by kosanna** [comment]
Moved out to a variable

**Comment by michaeltaranto** [comment]
I really like the new styles, looks a lot fresher. Playing around with the demo site I feel like we should have and active state that adds the accent border too. Without it there is a flicker effect when toggling the checkbox. What do you think?
http://seek-style-guide--14b0015b0121a313adbf09102621f220e54ac9b5.surge.sh/checkbox

File: react/Checkbox/Checkbox.less [hunk 2]
```diff
@@ -40,16 +40,16 @@
     .focus();
   }
 
-  .input:checked + .label & {
-    background-color: @sk-mid-gray-light;
+  .label & {
```
Comment (markdalgleish) [inline react/Checkbox/Checkbox.less]: Does this need to be nested anymore, since it doesn't depend on the `:checked` state?

**Comment by kosanna** [comment]
@michaeltaranto might lead to confusion between active and focused...thoughts?

File: react/Checkbox/Checkbox.less [hunk 3]
```diff
@@ -1,5 +1,14 @@
 @import (reference) "~seek-style-guide/theme";
 
+.activeFocusReset() {
+  .input:not([readonly]):not([disabled]):focus + .label &,
+  .input:not([readonly]):not([disabled]):active + .label & {
```
Comment (kosanna) [inline react/Checkbox/Checkbox.less]: New version of styling correctly handles this indentation. Older one wanted 4 spaces here

File: react/Header/UserAccount/UserAccount.less [hunk 4]
```diff
@@ -62,7 +63,7 @@
   .smallDeviceOnly({
     transform: translateY(@grid-row-height * -1);
     transition: opacity 0.25s ease, transform 0.25s ease;
-    top: (@grid-row-height * 7); // header height
+    top: @headerHeightMobile;
```
Comment (kosanna) [inline react/Header/UserAccount/UserAccount.less]: New version of styling complains on space after declaration, so instead of having a comment a decided to have a variable here, as it's anyway more correct. Scope creep

File: react/Header/Header.less [hunk 5]
```diff
@@ -1,5 +1,8 @@
 @import (reference) "~seek-style-guide/theme";
 
+@headerHeightDesktop: @grid-row-height * 9;
+@headerHeightMobile: @grid-row-height * 7;
```
Comment (markdalgleish) [inline react/Header/Header.less]: 👏 👏 👏

File: react/Header/UserAccount/UserAccount.less (hunk 4 above)
Comment (markdalgleish) [inline react/Header/UserAccount/UserAccount.less]: Yeah, this is actually a really nice cleanup 👍 

Thanks stylelint! 💯
